fastlane/lib/fastlane/lane.rb

```diff
@@ -42,9 +39,28 @@ def pretty_name
     end
 
     class << self
+      # Makes sure the lane name is valid
+      def verify_lane_name(name)
+        if self.black_list.include?(name.to_s)
+          UI.error "Lane Name '#{name}' can not be one of the followings: #{self.black_list}".red
+          UI.user_error!("Name '#{name}' is already taken")
+        end
+
+        if self.gray_list.include?(name.to_sym)
+          UI.error "The name '#{name}' for a lane equals the name of a fastlane tool"
```

Comment:
I think this is a little easier to understand and is closer to the black list message:
"Lane Name '#{name}' should not be used because it is the name of a fastlane tool"